File: CodeEval/easy/019bitpositions/solution.php
#!/usr/bin/env php
<?php

$fh = fopen($argv[1], "r");
while (($line = fgets($fh)) !== FALSE) {
    list($n, $i, $j) = explode(",", $line, 3);
    $n = (int) $n;
    $a = ($n >> ((int) $i-1)) & 1;
    $b = ($n >> ((int) $j-1)) & 1;
    echo (($a == $b) ? "true" : "false") . "\n";
}

File: codeeval/easy/019bitpositions/solution.lua
#!/usr/bin/env lua

for line in io.lines(arg[1]) do
  n = {}
  for str in line:gmatch("([^,]+)") do
    table.insert(n, tonumber(str))
  end

  if bit32.band(bit32.rshift(n[1], n[2]-1), 1) == bit32.band(bit32.rshift(n[1], n[3]-1), 1) then
    print("true")
  else
    print("false")
  end
end
